app/controllers/conversations.js

```diff
@@ -48,32 +73,63 @@ class ConversationsController extends BaseController {
         $or: [
           {
             type: "u",
-            owner_id: ObjectId(this.sessionRepository.getSessionUserId(ws)),
+            owner_id: ObjectId(currentUserId),
             opponent_id: conversationParams.opponent_id,
           },
           {
             type: "u",
             owner_id: ObjectId(conversationParams.opponent_id),
-            opponent_id: this.sessionRepository.getSessionUserId(ws),
+            opponent_id: currentUserId,
           },
         ],
       });
-      if (existingConversation)
+
+      if (existingConversation) {
+        const existingParticipants = await ConversationParticipant.findAll({
+          conversation_id: existingConversation._id,
+        });
+        if (existingParticipants.length !== 2) {
+          const requiredParticipantsIds = [
+            currentUserId,
+            conversationParams.opponent_id,
+          ];
+          const existingParticipantsIds = existingParticipants.map((u) =>
+            u.user_id.toString()
+          );
+
+          const missedParticipantId = requiredParticipantsIds.filter(
+            (uId) => !existingParticipantsIds.includes(uId)
+          )[0];
+
+          const participant = new ConversationParticipant({
+            user_id: ObjectId(missedParticipantId),
+            conversation_id: existingConversation._id,
+          });
+          await participant.save();
+
```

Comment:
looks good. Just one thing - let's retrieve user login here, not at the beginign of `create` action